windows/devices/uwb/include/windows/devices/uwb/IUwbDeviceDdi.hxx

```diff
@@ -32,46 +37,50 @@ struct IUwbDeviceDdi
     virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::optional<uint32_t>>>
     GetSessionCount() = 0;
 
-    // IOCTL_UWB_SESSION_INIT
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionInitialize(uint32_t sessionId, ::uwb::protocol::fira::UwbSessionType sessionType) = 0;
-
-    // IOCTL_UWB_SESSION_DEINIT
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionDeinitialize(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_GET_SESSION_STATE
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::optional<::uwb::protocol::fira::UwbSessionState>>>
-    SessionGetState(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_START_RANGING_SESSION
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionRangingStart(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_STOP_RANGING_SESSION
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionRangingStop(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_GET_RANGING_COUNT
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::optional<uint32_t>>>
-    SessionGetRangingCount(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_SESSION_UPDATE_CONTROLLER_MULTICAST_LIST
-    virtual std::future<::uwb::protocol::fira::UwbSessionUpdateMulicastListStatus>
-    SessionUpdateControllerMulticastList(uint32_t sessionId, ::uwb::protocol::fira::UwbMulticastAction multicastAction, std::vector<::uwb::UwbMacAddress> controlees) = 0;
-
-    // IOCTL_UWB_GET_APP_CONFIG_PARAMS
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameter>>>
-    GetApplicationConfigurationParameters(uint32_t sessionId, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameterType> applicationConfigurationParameterTypes) = 0;
-
-    // IOCTL_UWB_SET_APP_CONFIG_PARAMS
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::vector<::uwb::protocol::fira::UwbSetApplicationConfigurationParameterStatus>>>
-    SetApplicationConfigurationParameters(uint32_t sessionId, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameter> applicationConfigurationParameters) = 0;
-
     // TODO: unspecified IOCTLs below
     // IOCTL_UWB_SET_DEVICE_CONFIG_PARAMS
     // IOCTL_UWB_GET_DEVICE_CONFIG_PARAMS
 };
+
+struct IUwbDeviceDdiConnector : public IUwbDeviceDdi
+{
+    /**
+     * @brief Get the name of this device.
+     *
+     * @return const std::string&
+     */
+    virtual const std::string&
+    DeviceName() const noexcept = 0;

```

Comment:
This shouldn't be in the interface since it doesn't map to a DDI function. It can easily stay in `UwbDeviceConnector` though.

Reply:
I moved it into UwbDeviceConnector but I'm unsure how users would call it since they are supposed to either talk only with IUwbDeviceDdiConnector or IUwbSessionDdiConnector

Reply:
If moving it didn't break anything, that might be reason that it's not even needed. So, I wouldn't worry about it for now. Also, the thing that creates it will inherently know it's type, so if really necessary, it can do a `dynamic_cast` on the underlying pointer to get the real type instance.